test/primitives_test.dart

```diff
@@ -2,8 +2,108 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:ffi';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:ffi/ffi.dart';
+import 'package:native_synchronization/primitives.dart';
+import 'package:native_synchronization/sendable.dart';
 import 'package:test/test.dart';
 
 void main() {
-  test('dummy', () {});
+  group('mutex', () {
+    test('simple', () {
+      final mutex = Mutex();
+      mutex.lock();
+      mutex.unlock();
+    });
+
+    void spawnTestIsolate(int ptrAddress, Sendable<Mutex> sendableMutex) {
+      Isolate.run(() {
+        final ptr = Pointer<Uint8>.fromAddress(ptrAddress);
+        final mutex = sendableMutex.materialize();
+
+        while (true) {
+          sleep(Duration(milliseconds: 10));
+          if (mutex.runLocked(() {
+            if (ptr.value == 2) {
+              return true;
+            }
+            ptr.value = 0;
```

Comment:
The `calloc` makes it default initialize to `0` already. I'd write this code differently:
```
helperIsolate() {
  mutex.runLocked(() {
    ptr.value = 1;
    sleep(500 ms);
    ptr.value = 2;
  });
}

...
while (true) {
  await Future.delayed(10ms); // Not too busy loop.
  if (mutex.runLocked(() {
    if (ptr.value == 0) return false; // Helper hasn't initialized yet
    if (ptr.value == 2); return true; // Helper has initialized, blocked mutex for 500 ms & finished
    throw 'should never see any other value';
  }) break;
}
...
```